packages/web/hooks/useDebouncedCallback.ts

```diff
@@ -0,0 +1,31 @@
+import { useRef, useCallback } from 'react'
+
+/**
+ * Returns a memoized function that will only call the passed function when it hasn't been called for the wait period
+ * @param callback The function to be called
+ * @param wait Wait period after function hasn't been called for
+ * @returns A memoized function that is debounced
+ */
+const useDebouncedCallback = <CallbackArgs extends any[]>(
+  callback: (...args: CallbackArgs) => void,
+  wait: number,
+) => {
+  // Use a ref to store the timeout between renders
+  // and prevent changes to it from causing re-renders
+  const timeout = useRef()
+
+  return useCallback(
+    (...args) => {
+      const later = () => {
+        clearTimeout(timeout.current)
+        callback(...args)
+      }
+
+      clearTimeout(timeout.current)
+      timeout.current = setTimeout(later, wait)
+    },
+    [callback, wait],
```

Comment:
because `callback` here is in the dependency array, to achieve a stable value the caller will have to memoize their function (likely using `useCallback`) before passing it to this function, so we don't achieve much by the `useCallback` call.

Since callers will have to memoize anyway (and the `useRef` here can be accomplished with just an object declared outside the returned function) I'd suggest making this simply be a function rather than a hook that callers can use in a `useMemo`.